### PR TITLE
Support multi-valued ClasspathEntry attributes

### PR DESCRIPTION
**Pull request by jjohannes**
Implements the suggestion in #19529.

If the attribute value is an Iterable, an `<attribute>` node is produced for each element in the iterable.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- ~[ ] Provide unit tests (under `<subproject>/src/test`) to verify logic~
- ~[ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes~
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by cobexer** [comment]
Thank you for your contribution!

This PR has a valid DCO and tests. The relevant team for this area will confirm the design of the implementation choices.

File: platforms/ide/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/AbstractClasspathEntry.java [hunk 1]
```diff
@@ -188,10 +189,21 @@ public void writeEntryAttributes(Node node) {
 
         Node attributesNode = getAttributesNode(node, "attributes");
 
-        effectiveEntryAttrs.forEach((key, value) ->
-            attributesNode.appendNode("attribute", ImmutableMap.of(
-                "name", key,
-                "value", value)));
+        effectiveEntryAttrs.forEach((key, value) -> {
+            if (value instanceof Iterable) {
```
Comment (donat) [inline platforms/ide/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/AbstractClasspathEntry.java:193]: To increase readability, can you please add a comment explaining that we need this support mult-valued classpath entries?

**Review by donat — COMMENTED** [review]
LGTM, but I'd prefer a bit of documentation in the production code.

File: platforms/ide/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/AbstractClasspathEntry.java (hunk 1 above)
Comment (jjohannes) [reply platforms/ide/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/AbstractClasspathEntry.java]: Thanks @donat. I added a comment to the added code.

**Comment by donat** [comment]
@bot-gradle test RFM

File: platforms/ide/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/AbstractClasspathEntry.java (hunk 1 above)
Comment (donat) [reply platforms/ide/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/AbstractClasspathEntry.java]: Thank you. Let's see if CI has any complaints.

**Comment by donat** [comment]
@bot-gradle test and merge

**Comment by donat** [comment]
@jjohannes thank you for the contribution!

**Pull request merged**